opendevin/runtime/plugins/agent_skills/agentskills.py

```diff
@@ -147,37 +213,41 @@ def _print_window(CURRENT_FILE, CURRENT_LINE, WINDOW, return_str=False):
             print(output)
 
 
-def _cur_file_header(CURRENT_FILE, total_lines):
+def _cur_file_header(CURRENT_FILE, total_lines) -> str:
+    if not CURRENT_FILE:
+        return ''
     return f'[File: {os.path.abspath(CURRENT_FILE)} ({total_lines} lines total)]\n'
 
 
 @update_pwd_decorator
-def open_file(path: str, line_number: Optional[int] = None) -> None:
+def open_file(
+    path: str, line_number: int | None = 1, context_lines: int | None = 100
+) -> None:
     """
     Opens the file at the given path in the editor. If line_number is provided, the window will be moved to include that line.
 
     Args:
-        path: str: The path to the file to open.
-        line_number: Optional[int]: The line number to move to.
+        path: str: The path to the file to open, preferredly absolute path.
+        line_number: int | None = 1: The line number to move to. Defaults to 1.
+        context_lines: int | None = 100: The total number of lines to display in the window, with line_number as the center (if possible). Defaults to 100.
     """
-    global CURRENT_FILE, CURRENT_LINE
+    global CURRENT_FILE, CURRENT_LINE, WINDOW
+
     if not os.path.isfile(path):
         raise FileNotFoundError(f'File {path} not found')
 
     CURRENT_FILE = os.path.abspath(path)
     with open(CURRENT_FILE) as file:
-        total_lines = sum(1 for _ in file)
-
-    if line_number is not None:
-        if (
-            not isinstance(line_number, int)
-            or line_number < 1
-            or line_number > total_lines
-        ):
-            raise ValueError(f'Line number must be between 1 and {total_lines}')
-        CURRENT_LINE = line_number
-    else:
-        CURRENT_LINE = 1
+        total_lines = max(1, sum(1 for _ in file))
+
+    if not isinstance(line_number, int) or line_number < 1 or line_number > total_lines:
+        raise ValueError(f'Line number must be between 1 and {total_lines}')
+    CURRENT_LINE = line_number
+
+    # Override WINDOW with context_lines
+    if context_lines is None or context_lines < 1:
+        context_lines = 100
+    WINDOW = _clamp(context_lines, 1, 2000)
```

Comment:
Do you think if we need to display some warnings when agent input `context_lines > 2000`? 
Something like "Max context_lines supported is 2000, use scroll up/down to view the file if you want to see more"

If not, the agent may spend time in figuring out why `open_file` is not giving it more lines (though unlikely, but still possible)

Reply:
In my experience, the LLM often just ignores the fact, that it only gets 100 lines by default when using `open_file`.
Expanding the prompt as per your suggestion makes sense and should help it.